### PR TITLE
#2868 disabling features for non vaccine items

### LINKED ISSUE #2868: Disable VVM Status' for non-vaccine items
## Describe the bug

Currently with the vaccine module enabled, you can enter a VVM status for non-vaccine items in a supplier invoice/stocktake

### To reproduce

1. Go to an SI or stocktake with vaccine module enabled
2. Add an item and go to edit it

### Expected behaviour

Should only be able to edit VVM Status' for items which are vaccines

### Proposed Solution

Disabled the column for non-vaccine items in an SI. For stocktakes, the column should not be showing up anyway.

### Version and device info

- App version: VVM Development
- Tablet model: N/A
- OS version: N/A

### Additional context

N/A

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #2868 

## Change summary

- Adds some new disable logic for disabling VVM Status' and Breach column for rows which are not vaccines

## Testing

N/A

### Related areas to think about

N/A

### CODE REVIEW

File: src/database/utilities/createRecord.js [hunk 1]
```diff
@@ -656,10 +656,9 @@ const createItemBatch = (database, item, batchString, supplier) => {
 
     // An item has a restriction of the location type a location must be related to,
     // in order to be assigned to the item.
-    const thisStoreId = database.getSetting(SETTINGS_KEYS.THIS_STORE_ID);
     const itemStoreJoin = database
       .objects('ItemStoreJoin')
-      .filtered('itemId == $0 && storeId == $1', itemId, thisStoreId)[0];
+      .filtered('itemId == $0 && joinsThisStore == true', itemId)[0];
```
Comment (joshxg) [inline src/database/utilities/createRecord.js:661]: bug fix!

File: src/widgets/DataTable/DataTableRow.js [hunk 2]
```diff
@@ -232,21 +232,28 @@ const DataTableRow = React.memo(
...
 
             case COLUMN_TYPES.ICON: {
+              const { hasBreached, isVaccine } = rowData ?? {};
+
               const icons = {
                 chevron_right: ChevronRightIcon,
                 history: () => <HistoryIcon color={SUSSOL_ORANGE} />,
                 pencil: () => <PencilIcon color={SUSSOL_ORANGE} />,
-                breach: () => <HazardIcon color={SUSSOL_ORANGE} />,
+                breach: () => (hasBreached ? <HazardIcon color={SUSSOL_ORANGE} /> : null),
```
Comment (joshxg) [inline src/widgets/DataTable/DataTableRow.js:241]: just only showing the icon when the item has breached

File: src/widgets/FridgeDisplayInfo.js [hunk 3]
```diff
@@ -84,8 +84,8 @@ export const FridgeDisplayInfo = ({
           <DateRangeSelector
             initialStartDate={fromDate}
             initialEndDate={toDate}
-            onChangeToDate={onChangeFromDate}
-            onChangeFromDate={onChangeToDate}
+            onChangeToDate={onChangeToDate}
+            onChangeFromDate={onChangeFromDate}
```
Comment (joshxg) [inline src/widgets/FridgeDisplayInfo.js:88]: bug fix.. im a crazy muppet

File: src/widgets/DataTable/DataTableRow.js [hunk 4]
```diff
@@ -263,11 +270,19 @@ const DataTableRow = React.memo(
...
 
-            case COLUMN_TYPES.DROP_DOWN:
+            case COLUMN_TYPES.DROP_DOWN: {
+              const { isVaccine = false } = rowData ?? {};
+
+              const disabledConditions = {
+                [COLUMN_KEYS.CURRENT_VVM_STATUS]: !isVaccine,
+              };
+
+              const disabledVVMStatus = disabledConditions[columnKey] || isDisabled;
```
Comment (mark-prins) [inline src/widgets/DataTable/DataTableRow.js:280]: This ` || isDisabled` isn't necessary as it is checked when used in line `285`. I prefer splitting it out - and having `disabledVVMStatus` specific to VVM status

Comment (joshxg) [reply src/widgets/DataTable/DataTableRow.js]: Thanks! 6701530

**Pull request merged**